packages/sources/dlc-btc-por/src/endpoint/proof-of-reserves.ts

```diff
@@ -1,16 +1,60 @@
 import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
-import { EmptyInputParameters } from '@chainlink/external-adapter-framework/validation/input-params'
-import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation/input-params'
+import {
+  AdapterRequest,
+  SingleNumberResultResponse,
+} from '@chainlink/external-adapter-framework/util'
 import { config } from '../config'
 import { porTransport } from '../transport/proof-of-reserves'
+import { AdapterInputError } from '@chainlink/external-adapter-framework/validation/error'
+
+export const inputParameters = new InputParameters(
+  {
+    network: {
+      type: 'string',
+      description: 'The name of RPC network.',
+      required: true,
+    },
+    dlcContract: {
+      type: 'string',
+      description: 'Contract address to fetch vaults.',
+      required: true,
+    },
+  },
+  [
+    {
+      network: 'arbitrum',
+      dlcContract: '0x20157DBAbb84e3BBFE68C349d0d44E48AE7B5AD2',
+    },
+  ],
+)
 
 export type BaseEndpointTypes = {
-  Parameters: EmptyInputParameters
+  Parameters: typeof inputParameters.definition
   Response: SingleNumberResultResponse
   Settings: typeof config.settings
 }
 
 export const endpoint = new AdapterEndpoint({
   name: 'reserves',
+  inputParameters,
   transport: porTransport,
+  customInputValidation: (
+    req: AdapterRequest<typeof inputParameters.validated>,
+  ): AdapterInputError | undefined => {
+    const { network } = req.requestContext.data
+    const networkName = network.toUpperCase()
```

Comment:
the network specific env vars are pretty common now, do you think it's worth getting in a follow up ticket to formalize them in a utility function?

Reply:
Yeah I think it makes sense. Either find a way to support this in config, or like you suggested